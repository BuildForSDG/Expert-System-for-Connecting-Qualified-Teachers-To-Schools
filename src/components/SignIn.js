import React, { Component } from 'react';
import { NavLink, Link } from 'react-router-dom';
import { Button } from 'reactstrap';
import group from '../assets/group.png';
import { FiArrowLeft } from 'react-icons/fi';
import axios from 'axios';



class SignIn extends Component {
    constructor(props) {
        super(props);

        this.onChangeRegisterEmail = this.onChangeRegisterEmail.bind(this);
        this.onChangeRegisterPassword = this.onChangeRegisterPassword.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
        this.state = {
            email: '',
            password: '',
        }
    }

    
    onChangeRegisterEmail(e) {
        this.setState({email: e.target.value})
      }

      onChangeRegisterPassword(e) {
        this.setState({password: e.target.value})
      }

    onSubmit(e) {
        e.preventDefault();
        
        this.setState({ error: false });
        const user = 
        {  
            email: this.state.email,
            password: this.state.password
         };
        axios.post('http://localhost:8000/api/login',user,{
            headers: {'Content-Type' : 'application/json'}
        })
        .then(res => {
           
            localStorage.setItem('usertoken',res.data.token)
            console.log(res);
            if(res){
                this.props.history.push('/welcome')
            }
        })
           .catch(error => {
                this.refs.email.value = "";
                this.refs.password.value = "";
                this.setState({ err: true });
                console.log(error);
            });

            this.setState({email: '', password: ''})
    }

    onChange(e) {
        const { first_name, value } = e.target;
        this.setState({ [name]: value });
    }

    render() {
        let error = this.state.err;
        let msg = (!error) ? 'Login Successful' : 'Wrong Credentials';
        let name = (!error) ? 'alert alert-success' : 'alert alert-danger';
        return (
            <div className="login-page" >
                <div className="sign-in-header" >
                    <div className="panel-body">

                        {error != undefined && <div className={name} role="alert">{msg}</div>}
                    </div>
                    <div className="signIn">
                        <div className="first-div">
                            <Link to="/welcome" style={{ textDecoration: 'none', color: 'white' }} ><FiArrowLeft /></Link>
                        </div>
                        <div>
                            <p>Sign in</p>
                        </div>
                    </div>
                    <img className="group-icon" src={group} alt="group header" />
                </div>
                <div className="welcome-back">
                    <p>Welcome Back</p>
                </div>
                <div className="sign-in-form">
                    <p>Hello there, Sign in to continue!</p>
                    <Form onSubmit={this.onSubmit}>
                        <Form.Group controlId="Email">
                          

                            
                                <Form.Control placeholder="Email" ref="email"  type="email" value={this.state.email}  className="form-control"  onChange={this.onChangeRegisterEmail} required />
                            
                        </Form.Group>

                        <Form.Group controlId="Password">
                          

                           
                                <Form.Control placeholder="Password" ref="password" type="password"  className="form-control" value={this.state.password}  onChange={this.onChangeRegisterPassword} required />
                            
                        </Form.Group>

                        
                        <div className="form-group">
                                            
                                                <div className="checkbox">
                                                    <label>
                                                        <input type="checkbox" name="remember" /> Remember Me
                                                    </label>
                                                </div>
                                        
                                        </div>

                                        <div className="form-group">
                                            
                                                <Button type="submit" className="sign-in-button">
                                                    Login
                                                </Button>
                                                <div className="already-have-an-account"><p>Already have an account? <span><Link to="/welcome" style={{ textDecoration: 'none', color: '#3f2c90' }}>Sign up</Link></span></p></div>
                                                <li className="forget-password">
                                                    <Link to = "/" style={{ textDecoration: 'none', color: '#3f2c90' }}>Forgot Your Password?</Link>
                                                </li>
                                    
                                        </div>
                    </Form>
                   
                </div>
            </div>
        );
    }
}

export default SignIn;