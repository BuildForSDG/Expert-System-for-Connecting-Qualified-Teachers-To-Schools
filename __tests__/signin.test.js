import React from 'react';
import * as renderer from 'react-test-renderer';
import SignIn from '../src/components/SignIn';
import { MemoryRouter  as Router } from 'react-router-dom';


describe('Sign in component', () => {

	test('renders sign in page correctly', () => {
		let	tree = renderer.create(<Router><SignIn /></Router>).toJSON();
	  	expect(tree).toMatchSnapshot();
	});

});