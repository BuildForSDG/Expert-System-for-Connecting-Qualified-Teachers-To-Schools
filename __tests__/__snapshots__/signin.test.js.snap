// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in component renders sign in page correctly 1`] = `
<div
  className="login-page"
>
  <div
    className="sign-in-header"
  >
    <div
      className="signIn"
    >
      <div
        className="first-div"
      >
        <a
          href="/welcome"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="19"
              x2="5"
              y1="12"
              y2="12"
            />
            <polyline
              points="12 19 5 12 12 5"
            />
          </svg>
        </a>
      </div>
      <div>
        <p>
          Sign in
        </p>
      </div>
    </div>
    <img
      alt="group header"
      className="group-icon"
      src=""
    />
  </div>
  <div
    className="welcome-back"
  >
    <p>
      Welcome Back
    </p>
  </div>
  <div
    className="sign-in-form"
  >
    <p>
      Hello there, Sign in to continue!
    </p>
    <form>
      <input
        className="form-control"
        placeholder="Email"
        type="text"
      />
      <input
        className="form-control"
        placeholder="Password"
        type="password"
      />
    </form>
    <p
      className="forget-password"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "#3f2c90",
            "textDecoration": "none",
          }
        }
      >
        Forget Password?
      </a>
    </p>
    <button
      aria-label={null}
      className="sign-in-button btn btn-secondary"
      onClick={[Function]}
    >
      Sign in
    </button>
    <div
      className="already-have-an-account"
    >
      <p>
        Already have an account? 
        <span>
          <a
            href="/welcome"
            onClick={[Function]}
            style={
              Object {
                "color": "#3f2c90",
                "textDecoration": "none",
              }
            }
          >
            Sign up
          </a>
        </span>
      </p>
    </div>
  </div>
</div>
`;
